### PR TITLE
feat: eggignore

### LINKED ISSUE #3: Add .eggignore file and allow users to chose between the ignore file or the files field in egg.json
**Is your feature request related to a problem? If so, please describe.**
Right now the nest.land CLI publishes only the files given in the `files` field, in `egg.json`. Sometimes people have a lot of files imported from various directories and it's hard to keep track of all of them.

Also if you create a new file it is extra time to add it to the `files` field.

**Describe the solution you'd like**
The best solution would be to let the users choose between the `files` field or a `.eggignore` file. If the `.eggignore` file extists, the cli uses it instead of the `files` field.

**Describe alternatives you've considered**
An alternative already exists, using `*`

### LINKED ISSUE #26: [FEATURE] Upload all files if the files field is empty
**Is your feature request related to a problem? If so, please describe.**
A lot of developers are having problems with the files field. They just want to upload all of their files

**Describe the solution you'd like**
If the files field is empty or does not exist, upload all files in the folder

### PR DESCRIPTION
**Pull request by notfilippo**
first .eggignore draft

### CODE REVIEW

**Review by SteelAlloy — APPROVED** [review]
LGTM, good job on this one
Tested manually and worked as expected.

Two things:
- We should adapt the `init` command to reflect this change, tell the users that they can add a `.eggignore` file and/or change the request for files to optional
- Do we want to upload the `egg.json` config file by default ?

File: src/api/fetch.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { IModule, Module } from "../module.ts";
-import { ENDPOINT, apiFetch } from "./common.ts";
+import { apiFetch, ENDPOINT } from "./common.ts";
```
Comment (t8) [inline src/api/fetch.ts:2]: Love how you noticed this and alphabetized it

**Review by t8 — APPROVED** [review]
Tested as well. Great work @qu4k! The community will LOVE this. 

I agree with everything @oganexon had to say. Now we need some documentation for this!

File: src/api/fetch.ts (hunk 1 above)
Comment (notfilippo) [reply src/api/fetch.ts:2]: 😄

**Pull request merged**